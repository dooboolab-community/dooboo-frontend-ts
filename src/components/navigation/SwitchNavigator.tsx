import { BrowserRouter, Route, Switch } from 'react-router-dom';
import React, { ReactElement } from 'react';

import Intro from '../screen/Intro';
import Temp from '../screen/Temp';

function SwitchNavigator(): ReactElement {
  return (
    <BrowserRouter>
      <div style={{ textAlign: 'center' }}>
        <Switch>
          <Route exact={true} path="/" render={(): ReactElement => <Intro />} />
          <Route render={(): ReactElement => <Temp />} />
        </Switch>
      </div>
    </BrowserRouter>
  );
}

export default SwitchNavigator;
