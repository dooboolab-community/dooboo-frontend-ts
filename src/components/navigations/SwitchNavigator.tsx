import {BrowserRouter, Route, Routes} from 'react-router-dom';

import Intro from '../pages/Intro';
import {ReactElement} from 'react';
import Temp from '../pages/Temp';

function SwitchNavigator(): ReactElement {
  return (
    <BrowserRouter>
      <div
        style={{
          textAlign: 'center',
          flex: 1,
          width: '100vw',
        }}
      >
        <Routes>
          <Route path="/" element={<Intro />} />
          <Route path="*" element={<Temp />} />
        </Routes>
      </div>
    </BrowserRouter>
  );
}

export default SwitchNavigator;
