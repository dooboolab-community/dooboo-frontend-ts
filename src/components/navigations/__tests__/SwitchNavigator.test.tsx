import {RenderResult, render} from '@testing-library/react';

import React from 'react';
import RootProvider from '../../../providers/index';
import SwitchNavigator from '../SwitchNavigator';

const props = {};

const component = (
  <RootProvider initialThemeType="light">
    <SwitchNavigator {...props} />
  </RootProvider>
);

describe('[SwitchNavigator] rendering', () => {
  let testingLib: RenderResult;

  beforeEach(() => {
    testingLib = render(component);
  });

  it('should renders without crashing', () => {
    const {baseElement} = testingLib;

    expect(baseElement).toBeTruthy();
  });
});
