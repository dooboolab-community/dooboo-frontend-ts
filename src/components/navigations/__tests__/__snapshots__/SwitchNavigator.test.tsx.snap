// Vitest Snapshot v1

exports[`[SwitchNavigator] rendering > should renders without crashing 1`] = `
<body>
  <div>
    <div
      style="text-align: center; flex: 1; width: 100vw;"
    >
      <div
        class="css-1nvx4il"
      >
        <div
          class="css-1vx0ptc"
        >
          <span
            class="css-1l2m8ze"
          />
          <span
            class="css-1l2m8ze"
          />
          <span
            class="css-1l2m8ze"
          />
        </div>
        <div
          class="css-1j4jnb0"
        >
          <button
            class="css-wnig0z"
            data-testid="SIGN_IN"
            style="display: flex; height: 60px;"
          >
            <div
              class="css-qfy95"
            >
              <img
                class="css-1oidkpf"
                src="/assets/icons/ic-google-white@3x.png"
              />
              <span
                class="css-6apmke"
              />
            </div>
          </button>
          <button
            class="css-wnig0z"
            style="display: flex; height: 60px;"
          >
            <div
              class="css-qfy95"
            >
              <span
                class="css-6apmke"
              />
            </div>
          </button>
          <button
            class="css-wnig0z"
            data-testid="CHANGE_THEME"
            style="display: flex; height: 60px;"
          >
            <div
              class="css-qfy95"
            >
              <span
                class="css-6apmke"
              />
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
